src/middlewares/authorize.ts

```diff
@@ -0,0 +1,50 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { Request, Response, NextFunction } from "express";
+import jwt from "jsonwebtoken";
+import { UsersAttributes } from "../databases/models/users";
+import authRepository from "../modules/auth/repository/authRepositories";
+
+const SECRET: string = process.env.JWT_SECRET;
+
+interface ExtendedRequest extends Request {
+  user: UsersAttributes;
+}
+
+export const authorize = function (roles: string[]) {
+  return async (req: ExtendedRequest, res: Response, next: NextFunction) => {
+    try {
+      let token: string;
+      if (req.headers.authorization?.startsWith("Bearer")) {
+        token = req.headers.authorization.split(" ").at(-1);
+      }
+
+      if (!token) throw new Error("Login to get access to this resource");
+
+      const decoded: any = await jwt.verify(token, SECRET);
+
+      const user = await authRepository.findUserByAttributes("id", decoded.id);
+
+      if (!user) {
+        throw new Error("User belonging to this token does not exist");
+      }
+
+      if (!roles.includes(user.role)) {
+        throw new Error("You're not allowed to access this resource");
```

Comment:
Please follow the code convention we are using to return the response. And the message must be "Not authorized" with status of UNAUTHORIZED.